pylint/config/find_default_config_files.py

```diff
@@ -22,7 +22,8 @@
     Path(".pylintrc.toml"),
 )
 PYPROJECT_NAME = Path("pyproject.toml")
-CONFIG_NAMES = (*RC_NAMES, PYPROJECT_NAME, Path("setup.cfg"))
+TOX_NAME = Path("tox.ini")
```

Comment:
Why make this a constant? Can't it be like `setup.cfg`?

Reply:
Thanks @DanielNoord. Agreed.
While I have your attention, what's your opinion on [this logic](https://github.com/pylint-dev/pylint/blob/main/pylint/config/find_default_config_files.py#L64) and the associated comments around the inconsistencies mentioned on the issue? It does feel like we should allow `[pylint]` and not necessarily `[pylint.something]` but there could be some rationale for that.

Reply:
Yes, we previously only allowed `pylint.` because the way `optparse` worked required sections. I removed that when migrating to `argparse` and improving support for `toml`.

I would support changing this to be 1. exactly "pylint" or 2. anything that starts with "pylint.". To make sure we are not starting to parse `"pylint-internal-plugin"` etc.

Reply:
FYI @Pierre-Sassoulas and also may need to wait/further refactor before merging.

Reply:
Thanks for the ping :) I think anything starting with ``pylint.`` is closer to what's happening in pyproject.toml so I would favor that.

Reply:
In `.toml` we support `[tool.pylint]`. I think making `.ini` in line with this makes sense.

Reply:
Should we make the ``tool`` mandatory ? As in ``[tool.pylint]``, or ``[tool.pylint.main]``, ``[tool.pylint.basic]`` (etc.). We don't in ini file currently, but the ini file is pylint specific so it doesn't need the "pylint" / "tool.pylint" part.

https://github.com/pylint-dev/pylint/blob/2ba88b9317ff37a1e94007d00bbcad65ed515e18/examples/pylintrc#L119

So maybe  ``[pylint]``, or ``[pylint.main]``, ``[pylint.basic]``. for consistency in tox.ini (consistency with file of the same format above consistency with toml), and backward compatibility in ini ?

Reply:
`tool` is a thing because of the `pyproject.toml` spec I think.

Seeing other `tox.ini` files I think nobody would be surprised that `[pylint]` would be parsed as the configuration for `pylint`. We currently require people to do`[pylint.some_section_that_isnt_even_validated]` which feels cumbersome and inconsistent. That's why I proposed changing it from only `[pylint\..+]` to also allow `[pylint]`.

Reply:
Ok thanks folks! I've  tried to match Daniel's vision with the latest commit; accept `[pylint]` or `[pylint.<something_arbitrary>]`. Let me know if I've gotten anything wrong!